sidecar/supervisor/appsupervisor.go

```diff
@@ -118,8 +129,23 @@ func (a *AppSupervisor) DoAppSupervision(agent *register.RegistrationAgent) {
 					log.Errorf("App failed to start: %v", err)
 				}
 			}
-			exit(a.helpers, syscall.SIGKILL)
-			a.Shutdown(exitCode)
+
+			switch err.Proc.Action {
+			case config.DoNothingOnFailrue:
+				//Ignore this dead process
+				log.WithError(err.Err).Warn("App '%s' with args '%s' exited with error.  Ignoring", err.Proc.Cmd.Args[0], strings.Join(err.Proc.Cmd.Args[1:], " "))
+				err.Proc.Cmd.Wait()
+
+			case config.RestartOnFailure:
+				//TODO add restart functionality
```

Comment:
If restart is not implemented, I'd suggest to remove it from the configuration, and open another issue to implement it.